wolfssl/wolfcrypt/settings.h

```diff
@@ -3586,6 +3586,11 @@ extern void uITRON4_free(void *p) ;
     #endif
 #endif
 
+
+#if (defined(HAVE_DSA) || !defined(NO_DSA)) && defined(NO_SHA)
```

Comment:
Again, `HAVE_DSA` isn't really needed. Just gating on `!defined(NO_DSA)` should be enough. And yes I had a typo in my example `!defined(NO_SHA)` -> `defined(NO_SHA)`.